pkg/controller/deployment/deployment_controller.go

```diff
@@ -64,24 +63,71 @@ func (d *DeploymentController) reconcileDeployments() error {
 }
 
 func (d *DeploymentController) reconcileDeployment(deployment *experimental.Deployment) error {
-	targetedRCs, err := d.getTargetedRCs(deployment)
+	switch deployment.Spec.Strategy.Type {
+	case experimental.RecreateDeploymentStrategyType:
+		return d.reconcileRecreateDeployment(*deployment)
+	case experimental.RollingUpdateDeploymentStrategyType:
+		return d.reconcileRollingUpdateDeployment(*deployment)
+	}
+	return fmt.Errorf("Unexpected deployment strategy type: %s", deployment.Spec.Strategy.Type)
+}
+
+func (d *DeploymentController) reconcileRecreateDeployment(deployment experimental.Deployment) error {
+	// TODO: implement me.
+	return nil
+}
+
+func (d *DeploymentController) reconcileRollingUpdateDeployment(deployment experimental.Deployment) error {
+	desiredRC, err := d.getDesiredRC(deployment)
 	if err != nil {
 		return err
 	}
-	desiredRC, err := d.getDesiredRC(deployment)
+
+	targetedRCs, err := d.getTargetedRCs(deployment)
 	if err != nil {
 		return err
 	}
-	// TODO: Scale up and down the targeted and desired RCs.
-	// For now, just print their names, until we start doing something useful.
-	for _, targetedRC := range targetedRCs {
-		glog.Infof("TargetedRC: %s", targetedRC.ObjectMeta.Name)
+	targetedPodsCount := d.getReplicaCountForRCs(targetedRCs)
+
+	canScaleUp := desiredRC.Spec.Replicas != deployment.Spec.Replicas
+	canScaleDown := targetedPodsCount != 0
```

Comment:
Why does the logic for canScaleUp and canScaleDown live outside of d.scaleUp and d.scaleDown? Can you just call d.scaleUp/d.scaleDown and encapsulate the logic of whether you should actually perform it inside those calls?